### ISSUE TITLE
Basic usage fails to compile on SBCL 2.1.7

### ISSUE THREAD

**Issue by mfiano** [alex-gutev/static-dispatch#5]
Trying this library out for the first time, and cannot get this basic usage to compile on SBCL 2.1.7:

```lisp
(in-package #:cl-user)

(defpackage #:origin2
  (:use #:static-dispatch-cl))

(in-package #:origin2)

(defstruct (aggregate
            (:predicate nil)
            (:copier nil))
  (storage (make-array 0 :element-type 'single-float :initial-element 0.0) :type (simple-array single-float (*))))

(defgeneric ref (sequence index))

(defmethod ref ((sequence aggregate) index)
  (elt (aggregate-storage sequence) index))

(defgeneric (setf ref) (value sequence index))

(defmethod (setf ref) (value (sequence aggregate) index)
  (setf (elt (aggregate-storage sequence) index) value))

(defun test ()
  (declare (optimize (speed 3) (safety 2)))
  (let ((a (make-aggregate :storage (make-array 3 :element-type 'single-float :initial-contents '(1.0 2.0 3.0)))))
    (declare (type aggregate a))
    (ref a 2)))
    
; file: /tmp/slimegXmN46
; in: DEFUN TEST
;     (ORIGIN2::REF ORIGIN2::A 2)
; 
; caught ERROR:
;   don't know how to dump #<STANDARD-METHOD ORIGIN2::REF (AGGREGATE T) {1001A3DC23}> (default MAKE-LOAD-FORM method called).
; 
; compilation unit finished
;   caught 1 ERROR condition

```

Edit: Reproducible with latest Quicklisp dist version of static-dispatch *as well as* tip of master

**Comment by alex-gutev** [comment]
This is a bug. The raw method objects are inserted into the inlined CALL-NEXT-METHOD function of the last applicable method. SBCL obviously doesn't know how to dump these hence the error. I will fix it.

**Comment by alex-gutev** [comment]
Should be fixed in commit ab5cf1b.

**Comment by mfiano** [comment]
Thank you, that fixed it.
